diff --git a/node_modules/@opcat-labs/opcat/cjs/publickey.cjs b/node_modules/@opcat-labs/opcat/cjs/publickey.cjs
index 59283a4..6bd902e 100644
--- a/node_modules/@opcat-labs/opcat/cjs/publickey.cjs
+++ b/node_modules/@opcat-labs/opcat/cjs/publickey.cjs
@@ -330,10 +330,10 @@ PublicKey.prototype.toBuffer = PublicKey.prototype.toDER = function () {
   });
 
   var prefix;
-  if (!this.compressed) {
-    prefix = Buffer.from([0x04]);
-    return Buffer.concat([prefix, xbuf, ybuf]);
-  } else {
+  // if (!this.compressed) {
+  //   prefix = Buffer.from([0x04]);
+  //   return Buffer.concat([prefix, xbuf, ybuf]);
+  // } else {
     var odd = ybuf[ybuf.length - 1] % 2;
     if (odd) {
       prefix = Buffer.from([0x03]);
@@ -341,7 +341,7 @@ PublicKey.prototype.toBuffer = PublicKey.prototype.toDER = function () {
       prefix = Buffer.from([0x02]);
     }
     return Buffer.concat([prefix, xbuf]);
-  }
+  // }
 };
 
 /**
diff --git a/node_modules/@opcat-labs/opcat/cjs/transaction/transaction.cjs b/node_modules/@opcat-labs/opcat/cjs/transaction/transaction.cjs
index e64b03f..09d9081 100644
--- a/node_modules/@opcat-labs/opcat/cjs/transaction/transaction.cjs
+++ b/node_modules/@opcat-labs/opcat/cjs/transaction/transaction.cjs
@@ -226,6 +226,10 @@ Transaction.prototype._getHash = function () {
   return Hash.sha256sha256(this.toBufferWriter(true).toBuffer());
 };
 
+Transaction.prototype.getId = function () {
+  return this.id;
+};
+
 /**
  * Retrieve a hexa string that can be used with bitcoind's CLI interface
  * (decoderawtransaction, sendrawtransaction)
@@ -1227,6 +1231,14 @@ Transaction.prototype.getEstimateSize = function () {
   return this._estimateSize();
 };
 
+/**
+ * Gets the virtual size of the transaction.
+ * @returns {number} The virtual size in bytes.
+ */
+Transaction.prototype.getVirtualSize = function () {
+  return this._estimateSize();
+};
+
 /**
  * Estimates the total size of the transaction in bytes.
  * Calculates the sum of:
diff --git a/node_modules/@opcat-labs/opcat/esm/publickey.js b/node_modules/@opcat-labs/opcat/esm/publickey.js
index c1b954a..61de022 100644
--- a/node_modules/@opcat-labs/opcat/esm/publickey.js
+++ b/node_modules/@opcat-labs/opcat/esm/publickey.js
@@ -330,10 +330,10 @@ PublicKey.prototype.toBuffer = PublicKey.prototype.toDER = function () {
   });
 
   var prefix;
-  if (!this.compressed) {
-    prefix = Buffer.from([0x04]);
-    return Buffer.concat([prefix, xbuf, ybuf]);
-  } else {
+  // if (!this.compressed) {
+  //   prefix = Buffer.from([0x04]);
+  //   return Buffer.concat([prefix, xbuf, ybuf]);
+  // } else {
     var odd = ybuf[ybuf.length - 1] % 2;
     if (odd) {
       prefix = Buffer.from([0x03]);
@@ -341,7 +341,7 @@ PublicKey.prototype.toBuffer = PublicKey.prototype.toDER = function () {
       prefix = Buffer.from([0x02]);
     }
     return Buffer.concat([prefix, xbuf]);
-  }
+  // }
 };
 
 /**
diff --git a/node_modules/@opcat-labs/opcat/esm/transaction/transaction.js b/node_modules/@opcat-labs/opcat/esm/transaction/transaction.js
index 72610c6..d50d6a6 100644
--- a/node_modules/@opcat-labs/opcat/esm/transaction/transaction.js
+++ b/node_modules/@opcat-labs/opcat/esm/transaction/transaction.js
@@ -787,6 +787,10 @@ Transaction.prototype.from = function (utxo) {
   return this;
 };
 
+Transaction.prototype.getId = function () {
+  return this.id;
+};
+
 /**
  * Adds an input to the transaction from a UTXO (Unspent Transaction Output).
  * Determines the appropriate input type (PublicKeyHashInput, PublicKeyInput, or generic Input)
@@ -1225,6 +1229,14 @@ Transaction.prototype.getEstimateSize = function () {
   return this._estimateSize();
 };
 
+/**
+ * Gets the virtual size of the transaction.
+ * @returns {number} The virtual size in bytes.
+ */
+Transaction.prototype.getVirtualSize = function () {
+  return this._estimateSize();
+};
+
 /**
  * Estimates the total size of the transaction in bytes.
  * Calculates the sum of:
diff --git a/node_modules/@opcat-labs/opcat/types/transaction/transaction.d.cts b/node_modules/@opcat-labs/opcat/types/transaction/transaction.d.cts
index 149abd9..7dc5d45 100644
--- a/node_modules/@opcat-labs/opcat/types/transaction/transaction.d.cts
+++ b/node_modules/@opcat-labs/opcat/types/transaction/transaction.d.cts
@@ -32,6 +32,7 @@ declare class Transaction {
     _sigType: number;
     sealed: boolean;
     readonly hash: any;
+    getId(): string
     readonly id: any;
     get inputAmount(): number;
     get outputAmount(): number;
@@ -394,6 +395,7 @@ declare class Transaction {
      * @returns {number} The estimated size in bytes.
      */
     getEstimateSize(): number;
+    virtualSize(): number
     private _estimateSize;
     private _removeOutput;
     /**
